src/hidimstat/marginal/partial_dependence_plot.py

```diff
@@ -0,0 +1,807 @@
+        A fitted estimator implementing predict, predict_proba, or decision_function.
+    method : str or list of str
+        Method to get model predictions: 'predict', 'predict_proba', or 'decision_function'.
+    Returns
+    -------
+    list of array
+        List of predictions for each grid point, each with shape:
+        - (n_samples,) for regression
+        - (n_samples, 1) for binary classification
+        - (n_samples, n_classes) for multiclass
+    """
+    predictions = []
+    for new_values in values:
+        _safe_assign(X, new_values, column_indexer=variable)
+
+        # Note: predictions is of shape
+        # (n_points,) for non-multioutput regressors
+        # (n_points, n_tasks) for multioutput regressors
+        # (n_points, 1) for the regressors in cross_decomposition (I think)
+        # (n_points, 1) for binary classification (positive class already selected)
+        # (n_points, n_classes) for multiclass classification
+        pred, _ = _get_response_values(est, X, response_method=method)
+        if len(pred.shape) > 1 and pred.shape[1] > 1:
+            pred = np.max(pred, axis=1)
+        predictions.append(np.squeeze(pred))
+    return predictions
+
+
+class PartialDependancePlot(BaseVariableImportance):
+    """
+    Partial Dependence Plot (PDP):footcite:t:`friedman2001greedy` for analyzing
+    feature effects on model predictions. This is based on individual conditional
+    expectation (ICE) :footcite:t:`goldstein2015peeking`.
+    PDP shows the average model prediction across different values of target features,
+    while marginalizing over the values of all other features. It helps understand
+    how features affect predictions on average. ICE curves show predictions for
+    individual samples as the feature value changes.
+    Feature importance scores are computed following :footcite:t:`greenwell2018simple`:
+    - For continuous features: standard deviation of PDP curve
+    - For categorical features: range of PDP values divided by 4
+    Parameters
+    ----------
+    estimator : BaseEstimator
+        A fitted estimator object implementing predict, predict_proba, or
+        decision_function. Must be a regressor or binary/multiclass classifier.
+        Multioutput-multiclass classifiers are not supported.
+    features : array-like of {int, str}, default=None
+        Features for which to compute partial dependence. Can be:
+        - Single feature: int or str
+        - Multiple features: list of int or str
+        Feature indices or names must match training data.
+    method : {'auto', 'predict_proba', 'decision_function'}, default='auto'
+        Method for getting predictions:
+        - 'auto': tries predict_proba first, falls back to decision_function
+        - 'predict_proba': uses predicted probabilities
+        - 'decision_function': uses decision function scores
+        Ignored for regressors which always use predict.
+    n_jobs : int, default=1
```

Comment:
I'd rather have n_jobs at the end of the signature

Reply:
done